### ISSUE TITLE
MissingNewLine triggered on builder chain like pattern

### ISSUE THREAD

**Issue by vanniktech** [pinterest/ktlint#380]
When using the `--experimental` flag with ktlint 0.31.0 and the following configuration:

```
[*.{kt,kts}]
indent_size=2
max_line_length=200
```

the following code is flagged:

```kotlin
    val shareEvent = ShareEvent()
        .putMethod(event.method)
        .putContentName(event.contentName)
        .putContentType(event.contentType)
```

This is the output:

`Missing newline after "="`

My code is perfectly intended. I don't know why I'd want to have the `ShareEvent()` on its own line.

**Comment by shashachu** [comment]
I think your first bug is the same one as this one https://github.com/pinterest/ktlint/issues/368 - i didn't know how to describe it properly, but an assignment followed by some other statements seems to trigger it.

**Comment by shashachu** [comment]
@vanniktech I'm going to close this one since it's a dupe of #368 - feel free to open a separate ticket for the parentheses case. I'm not sure yet whether these all have the same root cause or not.

**Comment by vanniktech** [comment]
Thanks. Once that other issue has a PR / fix, I can try it out again.
